src/states/DeckBuildingState.py

```diff
@@ -2,16 +2,133 @@
 from src.constants import *
 import pygame
+import sys
 
 class DeckBuildingState(BaseState):
+
+    def __init__(self):
+        super(DeckBuildingState, self).__init__()
+        self.avaliableCard = []
+        self.selectDeck = True
+        self.deckIndex = 0
+        self.avaliableCardIndex = 0
+        self.cardPerRow = 8
+        self.avaliableCardSpacing = 10
+        self.deckSpacing = 5
+        self.selectedCardSpacing = 20
+
+        # mock avaliable card
+        for card in card_dic.values():
+            self.avaliableCard.append(card)
+        
+
     def Enter(self, params):
-        pass
+        self.deck = params["deck"]
 
     def Exit(self):
         pass
 
     def update(self, dt, events):
-        pass
+        for event in events:
+            if event.type == pygame.QUIT:
+                pygame.quit()
+                sys.exit()
+            if event.type == pygame.KEYDOWN:
+                if event.key == pygame.K_RIGHT:
+                    if self.selectDeck:
+                        if (self.deckIndex)%self.cardPerRow == (self.cardPerRow - 1):
+                            self.selectDeck = False
+                        elif self.deckIndex < len(self.deck.card_deck) -1:
+                            self.deckIndex +=1 
+                if event.key == pygame.K_LEFT:
+                    if self.selectDeck:
+                        if self.deckIndex > 0:
+                            self.deckIndex -=1
+                    else:
+                        self.selectDeck = True
+                if event.key == pygame.K_DOWN:
+                    if self.selectDeck:
+                        if self.deckIndex + self.cardPerRow < len(self.deck.card_deck):
+                            self.deckIndex += self.cardPerRow
+                    else:
+                        if self.avaliableCardIndex < len(self.avaliableCard) - 1:
+                            self.avaliableCardIndex += 1
+                if event.key == pygame.K_UP:
+                    if self.selectDeck:
+                        if self.deckIndex - self.cardPerRow >= 0:
+                            self.deckIndex -= self.cardPerRow
+                    else:
+                        if self.avaliableCardIndex > 0:
+                            self.avaliableCardIndex -= 1
+                if event.key == pygame.K_SPACE:
+                    if self.selectDeck:
+                        if self.deckIndex == len(self.deck.card_deck) - 1:
+                            self.deckIndex -= 1
+                        card = self.deck.card_deck[self.deckIndex]
+                        self.avaliableCard.append(card)
+                        self.deck.removeCard(card)
+                    else:
+                        if self.avaliableCardIndex == len(self.avaliableCard)- 1:
+                            self.avaliableCardIndex -= 1
+                        card = self.avaliableCard.pop(self.avaliableCardIndex)
+                        self.deck.addCard(card)
```

Comment:
@rymTooo This DeckBuildingState still have bug where we put all cards to the deck it will error and exit the game.